### PR TITLE
fix(console.table): fixes for character width breaking the layout

### LINKED ISSUE #7955: `console.table()` doesn't handle emoji widths properly
### What version of Bun is running?

1.0.21+837cbd60d

### What platform is your computer?

Linux 6.6.9-native_amd-xanmod1-1 x86_64 unknown

### What steps can reproduce the bug?

1. `console.table([{"test": "✅"}, {"test": "✅"}, {"test": "✅"}, {"test": "✅"}])`
2. `console.table([{"test 1": "✅"}, {"test 2": "✅"}, {"test 3": "✅"}, {"test 4": "✅"}])`

### What is the expected behavior?

Non-broken tables

### What do you see instead?

1.
image

2.
image

### Additional information

Terminal is kitty with the Cartograph CF font

### PR DESCRIPTION
**Pull request by otgerrogla**
### What does this PR do?

Contains 3 fixes:
- Replaces the algorithm for determining which unicode chars are full-width, by an auto-generated switch based on [official Unicode data](https://www.unicode.org/Public/UCD/latest/ucd/EastAsianWidth.txt). Specifically, characters tagged by unicode as `'W'` (wide) and `'F'` (full-width) are set to have width 2, the rest to 1.
    - Fixes several cases of emojis such as ✅ breaking the layout (the algorithm was outdated)
    - Fixes #7955
- Adds zero-width space (formerly known as BOM) as character with width 0
- Fixes bug where if an object-row did not have value for a given column (`obj.get()` returned `JSValue.empty`), an extra space was appended which broke the table layout.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

Fixed the snapshot test for the third issue. For the other two, wrote a test but it cannot be automated because there's no way to measure the actual printed width. Visual output: (the test prints about 2000 of the most common unicode codepoints, was run on a macOS)

image

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I included a test for the new code, or an existing test covers it
- [x] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed
- [x] I wrote TypeScript/JavaScript tests and they pass locally (`bun-debug test test-file-name.test`)

### CODE REVIEW

File: test/js/bun/console/console-table.test.ts [hunk 1]
```diff
@@ -157,3 +157,24 @@ test("console.table json fixture", () => {
   expect(actualOutput).toMatchSnapshot();
   console.log(actualOutput);
 });
+
+test.skip("console.table character widths", () => {
+  // note: this test cannot be automated because cannot test printed witdhs consistently.
```
Comment (Electroid) [inline test/js/bun/console/console-table.test.ts:162]: Not sure I understand this, why can't we take a snapshot of the stdout?

Comment (ThatOneCalculator) [reply test/js/bun/console/console-table.test.ts:162]: Also, comment should be "widths" not "witdhs"

Comment (otgerrogla) [reply test/js/bun/console/console-table.test.ts:162]: The unicode characters that are printed will be always output the same bytes, so a snapshot would not help.
What the test should look at is how they are actually visually printed in a terminal/tty, but there's no easy way to do that, even more if it has to support running on CI without actual visual terminals

Comment (otgerrogla) [reply test/js/bun/console/console-table.test.ts:162]: Even so, anyways how it actually ends up printed is gonna depend on the font configured by the user, the fallback fonts in the OS, etc.
So the safest bet is to rely the Unicode specs on width

**Comment by Jarred-Sumner** [comment]
Thank you!

**Pull request merged**